src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1787,6 +1755,7 @@ RouterJob(Query *originalQuery, PlannerRestrictionContext *plannerRestrictionCon
 
 	Job *job = CreateJob(originalQuery);
 	job->partitionKeyValue = partitionKeyValue;
+	job->onDummyPlacement = replacePrunedQueryWithDummy;
```

Comment:
do we need to track this here? This property is not specific to a Job, and it's not even specific to a Task, since a Task can consist of a combination of dummy queries and real shards.

Reply:
also, seems this is always true?

Reply:
No we don't. I know this is not a good solution. The problem is that in case of local tables the shard id will be invalid but some of the codepath assumes that it is not. For example CitusBeginModifyScan, And when shardId is invalid we would assume that the shards are pruned to 0 shards, which is not the case with local tables. This is something I will think but do you already have an idea of what we can do?

Reply:
For this I used `containsOnlyLocalTable` field to distinguish between pruning all shards vs having only local tables